surefire-booter/src/main/java/org/apache/maven/surefire/booter/SystemUtils.java

```diff
@@ -88,7 +88,7 @@ public static boolean endsWithJavaPath( String jvmExecPath )
     public static File toJdkHomeFromJvmExec( String jvmExecutable )
```

Comment:
Do we have unit test for it ... ?
If not please create - it should be easy 
Unit test should coverage all cases described in java doc for this method.

Reply:
No, I didn't feel the need to add a new unit test since the existing unit test fails with NPE under the described conditions (i.e JAVA_HOME set to a directory 2 levels up from root, `/opt/openjdk-bin-11.0.11_p9` in my case).

However, I have added a system-independent unit test to showcase the situation.

Reply:
I feel the fix is more complete now. Thanks for the comment